### PR TITLE
Create socket_server.py

### PR DESCRIPTION
**Pull request by Anbu-A**

### CODE REVIEW

File: socket_server.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        client.send(msg.encode())
+
+        # starts new thread
+        start_new_thread(new_con, (client, addr, ))
+
+        # Close the connection with the client
+        #client.close()
+
+if __name__ == '__main__':
+    Main()
```
Comment (HokageM) [inline socket_server.py:78]: add new line at the end of the file

File: socket_server.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# librarys
```
Comment (HokageM) [inline socket_server.py:1]: Remove this comment.
Remove the most comments.

clean code: "let the code tell the story"

Use comments for example for member functions in classes.

Please descripe in README.md what the program does.

File: socket_server.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import socket
+from _thread import *
+
+
+# symbolical constants
+BUFFER_SIZE = 1024
```
Comment (HokageM) [inline socket_server.py:7]: Do not use global variables. (change this later)

File: socket_server.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            
+            # server receives message from a client and send it 
+            # to all other connected clients 
+            for clients in CONNECTED_CLIENTS:
+                if(clients != client):  
+                    clients.send((f"[{client_name}]:{client_msg}\n").encode())                
+            break
+
+
+# source: https://www.geeksforgeeks.org/socket-programming-python/
```
Comment (HokageM) [inline socket_server.py:40]: This is good for the README.md

File: socket_server.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+def Main():
+    # next create a socket object
```
Comment (HokageM) [inline socket_server.py:42]: create a class file server.py and create a class Server.
Put the connent of the main in member function off Server.
Read this: https://stackoverflow.com/questions/1984325/explaining-pythons-enter-and-exit
And user Server class like
`with server as Server:
         server.run()`

File: socket_server.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+# global variables
+CONNECTED_CLIENTS = []
+CLIENT_NAMES = {} 
+
+
+# function for threading
+# handles messages from client,
+# every client has its own thread with this function
+def new_con(client, addr):
```
Comment (HokageM) [inline socket_server.py:20]: member function off Server class

File: socket_server.py (hunk 3 above)
Comment (HokageM) [reply socket_server.py:7]: Later: set this over command line arguments with module argparse.

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (HokageM) [inline .vscode/settings.json:1]: delete this file

File: LICENSE.txt [hunk 8]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
```
Comment (HokageM) [inline LICENSE.txt:1]: delete this file

File: main.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import sys
+from server import Server
+
+# symbolical constants;
```
Comment (HokageM) [inline main.py:4]: delete this constants

File: setup.cfg [hunk 10]
```diff
@@ -0,0 +1,130 @@
+# This file is used to configure your project.
+# Read more about the various options under:
+# http://setuptools.readthedocs.io/en/latest/setuptools.html#configuring-setup-using-setup-cfg-files
+
+[metadata]
+name = Messanger-TCP_Server
+description = Add a short description here!
```
Comment (HokageM) [inline setup.cfg:7]: add short description here

File: setup.cfg [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+author = Anbu-A
+author_email = 82471581+Anbu-A@users.noreply.github.com
+license = MIT
+long_description = file: README.rst
+long_description_content_type = text/x-rst; charset=UTF-8
+url = https://github.com/pyscaffold/pyscaffold/
```
Comment (HokageM) [inline setup.cfg:13]: add repository url here

File: setup.cfg [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+# Require a min/specific Python version (comma-separated conditions)
+# python_requires = >=3.8
+
+# Add here dependencies of your project (line-separated), e.g. requests>=2.2,<3.0.
+# Version specifiers like >=2.2,<3.0 avoid problems due to API changes in
+# new major versions. This works if the required packages follow Semantic Versioning.
+# For more information, check out https://semver.org/.
+install_requires =
+    importlib-metadata; python_version<"3.8"
+
```
Comment (HokageM) [inline setup.cfg:50]: add python packages here

File: setup.cfg [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+[devpi:upload]
+# Options for the devpi: PyPI server and packaging tool
+# VCS export must be deactivated since we are using setuptools-scm
+no_vcs = 1
+formats = bdist_wheel
+
+[flake8]
+# Some sane defaults for the code style checker flake8
+max_line_length = 88
```
Comment (HokageM) [inline setup.cfg:114]: 88 -> 120

File: src/messanger_tcp_server/main.py [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+def run():
+    """Calls :func:`main` passing the CLI arguments extracted from :obj:`sys.argv`
+
+    This function can be used as entry point to create console scripts with setuptools.
+    """
+    main(sys.argv[1:])
+
+
+if __name__ == "__main__":
+    # ^  This is a guard statement that will prevent the following code from
```
Comment (HokageM) [inline src/messanger_tcp_server/main.py:50]: remove this comments

File: README.rst [hunk 15]
```diff
@@ -0,0 +1,21 @@
+====================
+Messanger-TCP_Server
+====================
+
+
+Add a short description here!
```
Comment (HokageM) [inline README.rst:6]: do this

File: README.rst [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+
+
+Description
```
Comment (HokageM) [inline README.rst:9]: description -> Usage

and add here

.. code-block:: none
<HERE OUTPUT FROM msg_sever --help>

File: README.rst [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+===========
+
+A longer description of your project goes here...
+
+
+.. _pyscaffold-notes:
+
+Note
```
Comment (HokageM) [inline README.rst:17]: remove notes

File: .vscode/settings.json (hunk 7 above)
Comment (Anbu-A) [reply .vscode/settings.json:1]: Can't delete this, if I do the import will no longer work. 
But i added it to the gitignore file

Comment (Anbu-A) [reply .vscode/settings.json:1]: Yeah, I think that will work!

File: setup.cfg (hunk 12 above)
Comment (Anbu-A) [reply setup.cfg:53]: Should I also add the modules from main.py? like argparse or logging? 

Also how do I right this in there: "from _thread import *" ?

**Pull request merged**